### PR TITLE
Triton naming

### PR DESCRIPTION
**Pull request by deadeyegoodwin**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by deadeyegoodwin** [comment]
@ellis-bigelow @rakelkar 
I'm not sure I got all the references or fixed them correctly. There are other mentions of TensorRT but some seem related to integration with TF and those should stay as TensorRT. It seems like TensorRTSpec should be renamed but not sure.

**Comment by deadeyegoodwin** [comment]
There is also a related PR for KubeFlow website: https://github.com/kubeflow/website/pull/1846

File: docs/samples/triton/README.md [hunk 1]
```diff
@@ -30,16 +30,16 @@ echo $CLUSTER_IP
 ```
 curl -H "Host: ${SERVICE_HOSTNAME}" http://${CLUSTER_IP}/api/status
 ```
-3. edit /etc/hosts to map the CLUSTER IP to tensorrt-simple-string.default.example.com
+3. edit /etc/hosts to map the CLUSTER IP to triton-simple-string.default.example.com
 4. run the client
 ```
-docker run -e SERVICE_HOSTNAME:$SERVICE_HOSTNAME -it --rm --net=host kcorer/tensorrtserver_client:19.05
+docker run -e SERVICE_HOSTNAME:$SERVICE_HOSTNAME -it --rm --net=host nvcr.io/nvidia/tritonserver:20.03-py3
```
Comment (yuzisun) [inline docs/samples/triton/README.md]:
```suggestion
docker run -e SERVICE_HOSTNAME:$SERVICE_HOSTNAME -it --rm --net=host nvcr.io/nvidia/tritonserver:20.03-py3-clientsdk
```

File: docs/samples/triton/README.md [hunk 2]
```diff
@@ -30,16 +30,16 @@ echo $CLUSTER_IP
...
 ./build/simple_string_client -u $SERVICE_HOSTNAME
```
Comment (yuzisun) [inline docs/samples/triton/README.md]:
```suggestion
./workspace/install/bin/simple_string_client -u $SERVICE_HOSTNAME
```

**Comment by ellis-bigelow** [comment]
Yeah we're definitely missing changes for https://github.com/kubeflow/kfserving/blob/master/pkg/apis/serving/v1alpha2/framework_tensorrt.go
I'd run `make test` after making the changes.

**Comment by yuzisun** [comment]
> Yeah we're definitely missing changes for https://github.com/kubeflow/kfserving/blob/master/pkg/apis/serving/v1alpha2/framework_tensorrt.go
> I'd run `make test` after making the changes.

Good point, gonna add a e2e test in my other triton example PR.

**Comment by deadeyegoodwin** [comment]
So do I need to make changes to framework_tensorrt.go, or @yuzisun are you going to handle that in your followup PR?

**Comment by yuzisun** [comment]
> So do I need to make changes to framework_tensorrt.go, or @yuzisun are you going to handle that in your followup PR?

@deadeyegoodwin I can take care of that in my PR, once I am done we can merge together.

**Comment by rakelkar** [comment]
Not sure this change makes sense to me? In my understanding KFServing "Frameworks" are ML frameworks - which inference server we use to inference them should be an implementation detail. 

Triton is an inference server, and we should certainly update the TensorRT framework to use it (and possibly some other or all other frameworks to use it as well e.g. #744 ) but I don't think Triton is an ML framework.

So it seems logical to me to include TensorRT as a framework rather (assuming NVIDIA cares enough to support it) and consider updating the TensorRT sample to use an actual TRT model.plan instead of TF as it currently does.

@ellis-bigelow @cliveseldon @animeshsingh  @yuzisun 

/hold

**Comment by ellis-bigelow** [comment]
@yuzisun , I'd lean away from specifying inferenceserver in framework. This multilayered model seems more complex than it's worth.

**Comment by rakelkar** [comment]
@yuzisun when we say ONNX - it has to be a model serialized in ONNX format.. it doesnt matter how you got to that point... and from user perspective it doesnt matter whether you use the C++ ONNXServer for inference or your own Python Server that hosts the ONNX runtime... that is a serving infra decision...

So IMO framework maps directly to the serialized model format...

**Comment by rakelkar** [comment]
> @yuzisun , I'd lean away from specifying inferenceserver in framework. This multilayered model seems more complex than it's worth.



> I agree with Rakesh's point. QQ though, is the TensorRT name unchanged re: the framework?

I believe they only changed the name of the inference server (which makes sense since it supports multiple model formats)...

**Comment by yuzisun** [comment]
> > @yuzisun , I'd lean away from specifying inferenceserver in framework. This multilayered model seems more complex than it's worth.
> 
> > I agree with Rakesh's point. QQ though, is the TensorRT name unchanged re: the framework?
> 
> I believe they only changed the name of the inference server (which makes sense since it supports multiple model formats)...

So when user want to serve a tensorflow model with triton inference server, how do we specify that on our spec?

**Comment by rakelkar** [comment]
> like user first specify ML framework and then choose which inference server to serve the model

btw you can achieve this today by changing the configmap to point to a different inference server for the framework - but it is platform-wide opinion and not per-model... I guess our stance is the platform chooses for the 90% ... and edge cases are covered via custom? This is what cloud providers do too btw (e.g. sagemaker)

wdyt?

**Comment by deadeyegoodwin** [comment]
TensorRT "inference accelerator" is still called "TensorRT": https://developer.nvidia.com/tensorrt

**Comment by yuzisun** [comment]
> > like user first specify ML framework and then choose which inference server to serve the model
> 
> btw you can achieve this today by changing the configmap to point to a different inference server for the framework - but it is platform-wide opinion and not per-model... I guess our stance is the platform chooses for the 90% ... and edge cases are covered via custom? This is what cloud providers do too btw (e.g. sagemaker)
> 
> wdyt?

I would agree with you, but we are not there yet I think, also currently it is not simply a change in configmap with the image for a given ML framework, the startup command and parameters are pretty hard coded for a given inference server in the controller. Even that could change in the future I would still think giving user the ability to choose which inference server to use per model is needed, like for pytorch what if the model can not be converted to onnx and user just want to serve on a simple KFServing python server without going custom route.

**Comment by rakelkar** [comment]
> model can not be converted to onnx and user just want to serve on a simple KFServing python server without going custom route

But for that we already support pytorch as a framework and have a PYT server (that loads native PYT serialized models and does not require ONNX)?

**Comment by yuzisun** [comment]
> > model can not be converted to onnx and user just want to serve on a simple KFServing python server without going custom route
> 
> But for that we already support pytorch as a framework and have a PYT server (that loads native PYT serialized models and does not require ONNX)?

ya just making an example, or a more relevant question regarding to triton inference server is that since it is a multi-framework inference server how can we allow user to serve tensorflow/pytorch model on triton inference server by leveraging their auto-batching, multi-model co-hosting features without having to convert to the accelerated model format TensorRT. Currently the only way to use triton inference server it to put under `tensorrt` spec but it is confusing and the model actually may not be in tensorRT format, what do you think we can proceed from here?

**Comment by ellis-bigelow** [comment]
I think model server might be a more accurate descriptor. If it's capable of serving many different model formats, triton definitely seems more clear than tensorrt to me.

**Comment by rakelkar** [comment]
> Can we simply treat triton as a slight exception to the framework rule for now and we can resolve this more holistically once we have a clearer understanding of where the industry is moving with model servers

Sounds fair! 

/unhold

**Comment by yuzisun** [comment]
oh needs to move the test model from `gs://kfserving-samples/models/tensorrt` to `gs://kfserving-samples/models/triton`

**Comment by ellistarn** [comment]
Are we planning another 0.3.x release before the v1beta1 API changes? Much of this renaming will happen there.

**Comment by edenbuaa** [comment]
> 
> 
> Are we planning another 0.3.x release before the v1beta1 API changes? Much of this renaming will happen there.

Also want to know.

File: docs/samples/triton/simple_string/triton.yaml [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
 metadata:
-  name: "tensorrt-simple-string"
+  name: "triton-simple-string"
 spec:
   default:
     predictor:
-      tensorrt:
-        storageUri: "gs://kfserving-samples/models/tensorrt"
+      triton:
+        storageUri: "gs://kfserving-samples/models/triton"
```
Comment (yuzisun) [inline docs/samples/triton/simple_string/triton.yaml:9]:
```suggestion
        storageUri: "gs://kfserving-samples/models/tensorrt"
```

Comment (yuzisun) [reply docs/samples/triton/simple_string/triton.yaml:9]: let's keep this bucket for now, I somehow lost the permission to move the bucket

File: docs/samples/triton/simple_string/README.md [hunk 4]
```diff
@@ -1,28 +1,28 @@
...
+# Predict on a InferenceService using Triton Inference Server
 ## Setup
 1. Your ~/.kube/config should point to a cluster with [KFServing installed](https://github.com/kubeflow/kfserving/blob/master/docs/DEVELOPER_GUIDE.md#deploy-kfserving).
 2. Your cluster's Istio Ingress gateway must be network accessible.
 
 ## Create the InferenceService
 Apply the CRD
 ```
-kubectl apply -f tensorrt.yaml 
+kubectl apply -f triton.yaml
```
Comment (yuzisun) [inline docs/samples/triton/simple_string/README.md:10]: need to rename the file under `simple_string` as well

**Comment by edenbuaa** [comment]
@yuzisun Could the kfserving v0.3.0 support triton server now?

**Pull request merged**